src/projectsCreate.ts

```diff
@@ -0,0 +1,117 @@
+import * as clc from "cli-color";
+
+import * as api from "./api";
+import * as FirebaseError from "./error";
+import * as logger from "./logger";
+import { pollOperation } from "./operation-poller";
+import { OraWrapper } from "./oraWrapper";
+
+const ONE_SECOND_MILLIS = 1000;
+
+export enum ParentResourceType {
+  ORGANIZATION = "organization",
+  FOLDER = "folder",
+}
+
+export interface ParentResource {
+  id: string;
+  type: ParentResourceType;
+}
+
+export async function createFirebaseProject(
+  projectId: string,
+  options: { displayName?: string; parentResource?: ParentResource }
+): Promise<{ projectId: string }> {
+  await createCloudProject(projectId, options);
+  const projectInfo = await addFirebaseToCloudProject(projectId);
+
+  logger.info("");
+  logger.info("🎉🎉🎉 Your Firebase project is ready! 🎉🎉🎉");
+  logger.info("");
+  logger.info("Project information:");
+  logger.info(`   - Project ID: ${clc.bold(projectInfo.projectId)}`);
+  logger.info(`   - Project Name: ${clc.bold(projectInfo.displayName)}`);
+  logger.info("");
+  logger.info("Firebase console is available at");
+  logger.info(`https://console.firebase.google.com/project/${clc.bold(projectId)}/overview`);
+  return projectInfo;
+}
+
+/**
+ * Send an API request to create a new Google Cloud Platform project and poll the LRO to get the
+ * new project information.
+ * @return {Promise} this function returns a promise that resolves to the new cloud project
```

Comment:
Having `{Promise}` (describing the type) in a JSDoc in a TS file is redundant and it can be removed (throughout).

Reply:
Removed type description